File: loop_transformations/sources/stripe.2.c
/*
* @@name:	stripe.2
* @@type:	C
* @@operation:	compile
* @@expect:	success
* @@version:	omp_6.0
*/
void func(float A[6][6])
{
  #pragma omp stripe sizes(2,2)
  for (int i = 0; i < 6; ++i)
    for (int j = 0; j < 6; ++j)
      A[i][j] = i+j;
}

void func_equivalent(float A[6][6])
{
  for (int i1 = 0; i1 < 2; i1+=1)
    for (int j1 = 0; j1 < 2; j1+=1)
      for (int i2 = i1; i2 < 6; i2+=2)
        for (int j2 = j1; j2 < 6; j2+=2)
          A[i2][j2] = i2+j2;
}

File: loop_transformations/sources/stripe.2.f90
! @@name:	stripe.2
! @@type:	F-free
! @@operation:	compile
! @@expect:	success
! @@version:	omp_6.0
subroutine func(A)
  implicit none
  real :: A(6,6)
  integer :: i, j

  !$omp stripe sizes(2,2)
  do i = 1, 6
    do j = 1, 6
      A(j,i) = i+j
    end do
  end do
end subroutine

subroutine func_equivalent(A)
  implicit none
  real :: A(6,6)
  integer :: i1, i2, j1, j2

  do i1 = 1, 2
    do j1 = 1, 2
      do i2 = i1, 6, 2
        do j2 = j1, 6, 2
          A(j2,i2) = i2+j2
        end do
      end do
    end do
  end do
end subroutine
